language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ImplicitNewExpressionNodeContext.java

```diff
@@ -70,7 +70,11 @@ public ImplicitNewExpressionNodeContext() {
 
         switch (node.parent().kind()) {
             case LISTENER_DECLARATION:
-                typeDescriptor = ((ListenerDeclarationNode) node.parent()).typeDescriptor();
+                ListenerDeclarationNode listenerDeclarationNode = (ListenerDeclarationNode) node.parent();
```

Comment:
Should we directly extract `((ListenerDeclarationNode) node.parent()).typeDescriptor()` instead?